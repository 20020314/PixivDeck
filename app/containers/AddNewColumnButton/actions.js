// @flow
import type { User } from '../../types/user'
import type { Action } from './actionTypes.js'
import { ADD_USER_ILLUST } from './constants'

export function addUserIllust(user: User): Action {
  return {
    type: ADD_USER_ILLUST,
    user,
  }
}
