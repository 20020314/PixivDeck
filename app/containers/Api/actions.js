// @flow
import type { Response } from 'services/api'

import { API_REQUEST_SUCCESS } from './actionTypes'
import type { ApiRequestSuccess } from './actionTypes'

export function apiRequestSuccess(response: Response): ApiRequestSuccess {
  return {
    type: API_REQUEST_SUCCESS,
    response,
  }
}
