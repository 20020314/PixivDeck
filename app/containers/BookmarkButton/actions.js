// @flow
import type { Restrict } from './types'

import {
  ADD_BOOKMARK_REQUEST,
  ADD_BOOKMARK_SUCCESS,
  ADD_BOOKMARK_FAILURE,
  DELETE_BOOKMARK_REQUEST,
  DELETE_BOOKMARK_SUCCESS,
  DELETE_BOOKMARK_FAILURE,
} from './actionTypes'
import type {
  AddBookmarkRequest,
  AddBookmarkSuccess,
  AddBookmarkFailure,
  DeleteBookmarkRequest,
  DeleteBookmarkSuccess,
  DeleteBookmarkFailure,
} from './actionTypes'

export function addBookmarkRequest(
  id: number,
  restrict: Restrict
): AddBookmarkRequest {
  return {
    type: ADD_BOOKMARK_REQUEST,
    id,
    restrict,
  }
}
export function addBookmarkSuccess(
  id: number,
  restrict: Restrict
): AddBookmarkSuccess {
  return {
    type: ADD_BOOKMARK_SUCCESS,
    id,
    restrict,
  }
}
export function addBookmarkFailure(
  id: number,
  error: string
): AddBookmarkFailure {
  return {
    type: ADD_BOOKMARK_FAILURE,
    id,
    error,
  }
}
export function deleteBookmarkRequest(id: number): DeleteBookmarkRequest {
  return {
    type: DELETE_BOOKMARK_REQUEST,
    id,
  }
}
export function deleteBookmarkSuccess(id: number): DeleteBookmarkSuccess {
  return {
    type: DELETE_BOOKMARK_SUCCESS,
    id,
  }
}
export function deleteBookmarkFailure(
  id: number,
  error: string
): DeleteBookmarkFailure {
  return {
    type: DELETE_BOOKMARK_FAILURE,
    id,
    error,
  }
}
