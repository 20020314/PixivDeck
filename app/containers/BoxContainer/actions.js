// @flow
import type { Action } from './actionTypes.js'
import { OPEN_PIXIV } from './constants'

export function openPixiv(id: number): Action {
  return {
    type: OPEN_PIXIV,
    id,
  }
}
