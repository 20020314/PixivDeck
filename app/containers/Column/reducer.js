// @flow
import update from 'utils/update'

export type BaseColumn = {
  illustIds: Array<number>,
  nextUrl: ?string,
}

type ActionTypes = {
  ADD_COLUMN_SUCCESS: $Subtype<string>,
  SET_NEXT_URL: $Subtype<string>,
}

export function baseReducer<
  T: BaseColumn,
  S: { [*]: T },
  AS: ActionTypes,
  A: { +type: $Subtype<string> }
>(types: AS, state: S, action: A): S {
  switch (action.type) {
    case types.SET_NEXT_URL:
      return update(state, action, { nextUrl: action.nextUrl })
    case types.ADD_COLUMN_SUCCESS:
      return update(state, action, { illustIds: [], nextUrl: null })
    default:
      return state
  }
}
