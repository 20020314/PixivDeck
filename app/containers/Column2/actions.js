// @flow
import type { Action } from './actionTypes.js'
import { ADD_COLUMN } from './constants'
import type { Body } from './reducer'

export function addColumn(id: string, body: Body): Action {
  return {
    type: ADD_COLUMN,
    id,
    body,
  }
}
