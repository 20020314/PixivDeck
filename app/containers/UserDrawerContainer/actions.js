// @flow
import type { Action } from './actionTypes.js'
import {
  FETCH_DRAWER_ILLUST,
  ADD_DRAWER_ILLUSTS,
  SET_NEXT_ILLUST_URL,
  SET_NEXT_MANGA_URL,
  ADD_DRAWER_USER,
  ADD_DRAWER_PROFILE,
  FETCH_USER_DETAIL,
  NEXT_DRAWER_PAGE,
} from './constants'
import type { User, Profile } from 'types/user'
import type { DrawerType } from './reducer'

export function fetchDrawerIllust(id: number, drawerType: DrawerType): Action {
  return {
    type: FETCH_DRAWER_ILLUST,
    id,
    drawerType,
  }
}

export function addDrawerIllusts(
  ids: number[],
  drawerType: DrawerType
): Action {
  return {
    type: ADD_DRAWER_ILLUSTS,
    ids,
    drawerType,
  }
}

export function setNextIllustUrl(url: string): Action {
  return {
    type: SET_NEXT_ILLUST_URL,
    url,
  }
}

export function setNextMangaUrl(url: string): Action {
  return {
    type: SET_NEXT_MANGA_URL,
    url,
  }
}

export function addDrawerUser(user: User): Action {
  return {
    type: ADD_DRAWER_USER,
    user,
  }
}

export function addDrawerProfile(profile: Profile): Action {
  return {
    type: ADD_DRAWER_PROFILE,
    profile,
  }
}

export function fetchUserDetail(id: number): Action {
  return {
    type: FETCH_USER_DETAIL,
    id,
  }
}

export function nextDrawerPage(drawerType: DrawerType): Action {
  return {
    type: NEXT_DRAWER_PAGE,
    drawerType,
  }
}
