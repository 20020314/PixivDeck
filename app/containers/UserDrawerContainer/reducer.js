// @flow
import union from 'lodash.union'
import type { User, Profile } from 'types/user'
import type { Action } from '../../action'
import * as ManegerActions from '../DrawerManager/constants'
import * as Actions from './constants'

export type DrawerType = 'illust' | 'manga'

export type State = {
  user: ?User,
  profile: ?Profile,
  illustList: number[],
  mangaList: number[],
  nextIllustUrl: ?string,
  nextMangaUrl: ?string,
  isLoading: boolean,
}

const initialState: State = {
  user: null,
  profile: null,
  illustList: [],
  mangaList: [],
  nextIllustUrl: null,
  nextMangaUrl: null,
  isLoading: false,
}

const setNextUrl = (type: DrawerType, nextUrl: string) => {
  return type === 'illust'
    ? { nextIllustUrl: nextUrl }
    : { nextMangaUrl: nextUrl }
}

function addDrawerIllusts(type: DrawerType, illusts: number[], ids: number[]) {
  if (type === 'manga') {
    return { mangaList: union(illusts, ids) }
  }
  return { illustList: union(illusts, ids) }
}

export default function(
  state: State = initialState,
  action: Action
): $Shape<State> {
  switch (action.type) {
    case ManegerActions.CLOSE_DRAWER:
      return initialState
    case Actions.ADD_DRAWER_USER:
      return { ...state, user: action.user }
    case Actions.ADD_DRAWER_PROFILE:
      return { ...state, profile: action.profile }
    case Actions.ADD_DRAWER_ILLUSTS:
      // todo 分割
      return {
        ...state,
        ...addDrawerIllusts(action.drawerType, state.illustList, action.ids),
      }
    case Actions.SET_NEXT_URL:
      return { ...state, ...setNextUrl(action.drawerType, action.url) }
    default:
      return state
  }
}
