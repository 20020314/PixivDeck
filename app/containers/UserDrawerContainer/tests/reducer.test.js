// @flow
import * as manegerActions from 'containers/DrawerManager/actions'
import reducer, { initialState as ds } from '../reducer'
import * as actions from '../actions'

test('default action', () => {
  // $FlowFixMe
  expect(reducer(ds, { type: 'default action' })).toMatchSnapshot()
})

test('CLOSE_DRAWER', () => {
  expect(reducer(ds, manegerActions.closeDrawer())).toMatchSnapshot()
})

test('ADD_DRAWER_USER', () => {
  // $FlowFixMe
  expect(reducer(ds, actions.addDrawerUser({ id: 123 }))).toMatchSnapshot()
})

test('ADD_DRAWER_PROFILE', () => {
  expect(
    // $FlowFixMe
    reducer(ds, actions.addDrawerProfile({ mock: 'mock' }))
  ).toMatchSnapshot()
})

test('Actions', () => {
  expect(actions.fetchIllustSuccess([1, 2, 3])).toMatchSnapshot()
})
