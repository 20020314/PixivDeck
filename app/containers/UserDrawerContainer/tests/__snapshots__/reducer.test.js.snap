// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADD_DRAWER_PROFILE 1`] = `
Object {
  "illustList": Array [],
  "isLoading": false,
  "mangaList": Array [],
  "nextIllustUrl": null,
  "nextMangaUrl": null,
  "profile": Object {
    "mock": "mock",
  },
  "user": null,
}
`;

exports[`ADD_DRAWER_USER 1`] = `
Object {
  "illustList": Array [],
  "isLoading": false,
  "mangaList": Array [],
  "nextIllustUrl": null,
  "nextMangaUrl": null,
  "profile": null,
  "user": Object {
    "id": 123,
  },
}
`;

exports[`Actions 1`] = `
Object {
  "ids": Array [
    1,
    2,
    3,
  ],
  "type": "UserDrawer/FETCH_ILLUST_SUCCESS",
}
`;

exports[`CLOSE_DRAWER 1`] = `
Object {
  "illustList": Array [],
  "isLoading": false,
  "mangaList": Array [],
  "nextIllustUrl": null,
  "nextMangaUrl": null,
  "profile": null,
  "user": null,
}
`;

exports[`default action 1`] = `
Object {
  "illustList": Array [],
  "isLoading": false,
  "mangaList": Array [],
  "nextIllustUrl": null,
  "nextMangaUrl": null,
  "profile": null,
  "user": null,
}
`;
